src/plugins/workspace/server/workspace_client.ts

```diff
@@ -392,12 +392,67 @@ export class WorkspaceClient implements IWorkspaceClientImpl {
       };
     }
   }
+
+  public async associate(
+    requestDetail: IRequestDetail,
+    workspaceId: string,
+    objects: Array<{ id: string; type: string }>
+  ): Promise<IResponse<Array<{ id: string; error?: string }>>> {
+    const savedObjectClient = this.getSavedObjectClientsFromRequestDetail(requestDetail);
+    const promises = objects.map(async (obj) => {
+      try {
+        await savedObjectClient.addToWorkspaces(obj.type, obj.id, [workspaceId]);
+        return {
+          id: obj.id,
+        };
+      } catch (e) {
+        return {
+          id: obj.id,
+          error: this.formatError(e),
+        };
+      }
+    });
+    const result = await Promise.all(promises);
+    return {
+      success: true,
+      result,
+    };
+  }
```

Comment:
`Promise.allSettled`?

Reply:
Could use `allSettled`, but it doesn't really matter  here as it already has try/catch inside each promise, no error will throw.